tests/e2e/performance/test_pvc_creation_deletion_performance.py

```diff
@@ -0,0 +1,291 @@
+        push_to_pvc_time_dashboard(self.interface, "1-pvc-creation", creation_average)
+        push_to_pvc_time_dashboard(self.interface, "1-pvc-deletion", deletion_average)
+
+    def write_file_on_pvc(self, pvc_obj, filesize=10):
+        """
+        Writes a file on given PVC
+        Args:
+            pvc_obj: PVC object to write a file on
+            filesize: size of file to write (in GB)
+
+        Returns:
+            Pod on this pvc on which the file was written
+        """
+        pod_obj = self.pod_factory(
+            interface=self.interface, pvc=pvc_obj, status=constants.STATUS_RUNNING
+        )
+
+        # filesize to be written is always 10 GB
+        file_size = f"{int(filesize * 1024)}M"
+
+        log.info(f"Starting IO on the POD {pod_obj.name}")
+        # Going to run only write IO
+        pod_obj.fillup_fs(size=file_size, fio_filename=f"{pod_obj.name}_file")
+
+        # Wait for fio to finish
+        fio_result = pod_obj.get_fio_results()
+        err_count = fio_result.get("jobs")[0].get("error")
+        assert (
+            err_count == 0
+        ), f"IO error on pod {pod_obj.name}. FIO result: {fio_result}"
+        log.info("IO on the PVC has finished")
+        return pod_obj
+
+    @pytest.mark.usefixtures(base_setup.__name__)
+    def test_multiple_pvc_deletion_measurement_performance(self, teardown_factory):
+        """
+        Measuring PVC deletion time of 120 PVCs in 180 seconds
+
+        Args:
+            teardown_factory: A fixture used when we want a new resource that was created during the tests
+                               to be removed in the teardown phase.
+        Returns:
+
+        """
+        number_of_pvcs = 120
+        pvc_size = "1Gi"
+        msg_prefix = f"Interface: {self.interface}, PVC size: {pvc_size}."
+
+        log.info(f"{msg_prefix} Start creating new 120 PVCs")
+
+        pvc_objs = helpers.create_multiple_pvcs(
+            sc_name=self.sc_obj.name,
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+            number_of_pvc=number_of_pvcs,
+            size=pvc_size,
+        )
+
+        for pvc_obj in pvc_objs:
+            pvc_obj.reload()
+            teardown_factory(pvc_obj)
+        with ThreadPoolExecutor(max_workers=5) as executor:
+            for pvc_obj in pvc_objs:
+                executor.submit(
+                    helpers.wait_for_resource_state, pvc_obj, constants.STATUS_BOUND
+                )
+
+                executor.submit(pvc_obj.reload)
+
+        pod_objs = []
+        for pvc_obj in pvc_objs:
+            pod_obj = self.write_file_on_pvc(pvc_obj, 0.3)
+            pod_objs.append(pod_obj)
+
+        # Get pvc_name, require pvc_name to fetch deletion time data from log
+        threads = list()
+        for pvc_obj in pvc_objs:
+            process = threading.Thread(target=pvc_obj.reload)
+            process.start()
+            threads.append(process)
+        for process in threads:
+            process.join()
```

Comment:
are you sure running 120 threads simultaneously will be faster ?

Reply:
@Avilir  I don't know, but this is how it worked before. So I thought that the essense of this test is to create and delete in threads. I did not want to change that, only added writing data before the deletion.